yt_dlp/extractor/piapro.py

```diff
@@ -0,0 +1,97 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+from .common import InfoExtractor
+from ..compat import compat_urlparse
+from ..utils import int_or_none, parse_duration, parse_filesize, unified_timestamp, urlencode_postdata, ExtractorError
+
+
+class PiaproIE(InfoExtractor):
+    _NETRC_MACHINE = 'piapro'
+    _VALID_URL = r'https?://piapro\.jp/t/(?P<id>\w+)/?'
+    _TESTS = [{
+        'url': 'https://piapro.jp/t/NXYR',
+        'md5': 'a9d52f27d13bafab7ee34116a7dcfa77',
+        'info_dict': {
+            'id': 'NXYR',
+            'ext': 'mp3',
+            'uploader': 'wowaka',
+            'uploader_id': 'wowaka',
+            'title': '裏表ラバーズ',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        }
+    }]
+
+    def _real_initialize(self):
+        self._login_status = self._login()
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if not username:
+            return False
+        login_ok = True
+        login_form_strs = {
+            '_username': username,
+            '_password': password,
+            '_remember_me': 'on',
+            'login': 'ログイン'
+        }
+        self._request_webpage('https://piapro.jp/login/', None)
+        urlh = self._request_webpage(
+            'https://piapro.jp/login/exe', None,
+            note='Logging in', errnote='Unable to log in',
+            data=urlencode_postdata(login_form_strs))
+        if urlh is False:
+            login_ok = False
+        else:
+            parts = compat_urlparse.urlparse(urlh.geturl())
+            if parts.path != '/':
+                login_ok = False
+        if not login_ok:
+            self._downloader.report_warning(
+                'unable to log in: bad username or password')
+        return login_ok
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        category_id = self._search_regex(r'categoryId=(.+)">', webpage, 'category ID')
+        if category_id not in ('1', '2', '21', '22', '23', '24', '25'):
+            raise ExtractorError('The URL does not contain audio.', expected=True)
+
+        str_duration, str_filesize = self._search_regex(
+            r'サイズ：</span>(.+?)/\(([0-9,]+?[KMG]?B)）', webpage, 'duration and size',
+            group=(1, 2), default=(None, None))
+        str_viewcount = self._search_regex(r'閲覧数：</span>([0-9,]+)\s+', webpage, 'view count', fatal=False)
+
+        str_filesize = str_filesize.replace(',', '')
```

Comment:
`parse_filesize('1,234MB')` (where the comma is for decimal separator, not decimal point) is treated like `1.234 MB` because of L2265 in utils.py. Also, `int()` doesn't work with commas. Still needed.